pkg/operator/certrotation/annotations.go

```diff
@@ -0,0 +1,37 @@
+package certrotation
+
+import (
+	"github.com/openshift/api/annotations"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewObjectMeta(name, namespace, jiraComponent, description string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Namespace: namespace,
+		Name:      name,
+		Annotations: map[string]string{
+			annotations.OpenShiftComponent:   jiraComponent,
+			annotations.OpenShiftDescription: description,
+		},
+	}
+}
+
+// EnsureTLSMetadataUpdate mutates objectMeta setting necessary annotations if unset
+func EnsureTLSMetadataUpdate(meta *metav1.ObjectMeta, jiraComponent, description string) bool {
+	modified := false
+	if meta.Annotations == nil {
+		meta.Annotations = map[string]string{
+			annotations.OpenShiftComponent:   "",
```

Comment:
followup, if you're lazy defining below, this leaves empty values distinct from unspecified.